# *Xylella fastidiosa* (Xf) MLST phylogeny
(created 09/2020)  
Pipeline to create a phylogeny using 293 publicly available Xf MLSTs from PubMLST. Also includes *Xylella taiwanensis* as outgroup and four Colombian strains which subspecies have been confirmed.

## Get sequences.

### 1. Download concatenated Xf MLST sequences from PubMLST and unzip all downloaded sequences.

### 2. Concatenate all FASTA files into a single large FASTA file.
```bash
cat *.fasta > xf_mlst_all.cat.fasta
```

### 3. Change FASTA header to isolate ID only.
This first script removes all characters from the beginning of the line until the first ```|``` occurence and replaces it with ```>```. The second script removes all characters after (and including) the first ```|``` occurence.
```bash
sed 's/[^|]*|/>/' xf293_mlst.cat.fasta > xf_mlst_all.cat.fasta
sed 's/[|].*//' xf293_mlst.cat.fasta > xf_mlst_all.cat.fasta
```

### 4. Add Colombian strains and *X. taiwanensis* (outgroup) to FASTA file.
```bash
cat RUBCA*.fasta >> xf_mlst_all.cat.fasta
cat xylella_taiwanensis.fasta >> xf_mlst_all.cat.fasta
```

### 5. Run Tcoffee alignment on FASTA file. Change the FASTA file directory within the sbatch script.
```bash
mkdir /data/data2/scratch2/mirabl/Xf_proj/MLST_Xf293/ALN
cd /data/data2/scratch2/mirabl/Xf_proj/MLST_Xf293/ALN
sbatch /home/mirabl/SUB_SLURM/tcoffee_sub.sh
```

### 4. Correct alignments using GBlocks. (what input format is required?)
```
cd ~
for file in /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/*.fasta; do
  Gblocks $file -t=p -d=y
  echo $file
done
```

### 25. Run the protein model tester on alignment. *****
```
cd /home/mirabl/
for file in $(cat /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Orthogroups_SingleCopyOrthologues2.txt); do
  echo $file
  perl /home/hulinm/git_repos/tools/analysis/python_effector_scripts/alignment_convert.pl -i /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/"$file" -o /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/"$file".phy -f phylip -g fasta
done
```

### 26. Test protein models for each alignment.
```
cd /home/mirabl/
for file in /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/*.phy; do
  file_short=$(basename $file | sed s/".phy"//g )
  echo $file_short Jobs=$(qstat | grep 'prottest.p' | wc -l);
    while [ $Jobs -gt 50 ]; do
      sleep 10;
      printf ".";
      Jobs=$(qstat | grep 'prottest.p' | wc -l)
    done
  qsub /home/mirabl/SUB_PBS/Xf_proj/prottest.pbs "$file" /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/"$file_short"_model
done
```

### 27. Get best model name into its own file.
```
mkdir /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/Model2/
for file in /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/*_model; do
  file_short=$(basename $file | sed s/"_model"//g)
  grep -i "Best model according to LnL" $file | cut -d " " -f6 > /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/Model2/"$file_short"
done
```

### 28. Move model files.
```
mkdir /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/Prottest_model2
for file in /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/*_model; do
  file_short=$(basename $file | sed s/"_model"//g)
  mv $file /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/Prottest_model2/"$file_short"
done
```

### 29. Proteins.
```
for file in $(cat /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/list); do
  cat /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/Model2/"$file" >> /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/Fasta/Single_copy/Align/Model2/models
done
```

### 30. Add sequence evolution model.
Make the final partition file.
```
cd /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_final/OrthoFinder/Formatted/OrthoFinder/Results_Jul04/Orthogroups/
mkfifo pipe1
mkfifo pipe2
#Add effector names in first column
cut -f1 Fasta/Single_copy/Align/Model2/models > pipe1 & cut -f1,2,3 Fasta/Single_copy/Align/positions2 > pipe2 & paste pipe1 pipe2 > Fasta/Single_copy/Align/partition2
mv pipe* /data2/scratch2/mirabl/Discard
sed s/"\t"/", "/g Fasta/Single_copy/Align/partition2 > Fasta/Single_copy/Align/partition_file2
```

### 32. Run IQTREE.
```
qsub /home/mirabl/SUB_PBS/Xf_proj/iqtree.pbs /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_w_outgr/OrthoFinder/Formatted/OrthoFinder/Results_May31/Orthogroups/Fasta/Single_copy/Align/combined.phy /data2/scratch2/mirabl/Xf_proj/NCBI_Xf55/Analysis_w_outgr/OrthoFinder/Formatted/OrthoFinder/Results_May31/Orthogroups/Fasta/Single_copy/Align/partition_file
